pkg/exporter/azureblob_exporter.go

```diff
@@ -32,6 +32,10 @@ func NewAzureBlobExporter(creationTime, hostname string) *AzureBlobExporter {
 	}
 }
 
+func (exporter *AzureBlobExporter) GetName() string {
```

Comment:
Why do we need GetName for exporters?

Reply:
so we can use it to select them, currently only one exporter (azureblob) but I have PR in the wings for localmachine as well.

Reply:
I do hope we are not going to select any structure based on a display name :D

Reply:
why not? This is perfect for the interim - as I mentioned in initial description, a better idea is to use a type registry and remove the need for GetName to be defined at all, but this is overkill for now.

Using the display name means we are not burdened with the unnecessary complexity of introducing an intermediate concept whose only purpose is to map from one name to another, and which we would have to refactor to remove if and when we move to a type registry anyway.

Reply:
There is no need to select exporters for now anyway so I believe this is just creating noise. I'd rather come up with an elegant solution the day we need to do this.

Reply:
well - the day we need to do this is today :) this PR would have included the motivation for doing this (the localMachineExporter) but on request I split the original PR up into multiple smaller PR, and haven't submitted the localMachineExporter one yet as I only want to have to rebase in one direction.

How is it inelegant to be doing this in the same way as collectors and diagnosers, and without having to introduce any new concepts? The pre-existing COLLECTOR_LIST is an inelegant solution by comparison as it introduces another layer of indirection in the middle